plugins/search-react/src/components/SearchResultGroup/SearchResultGroup.tsx

```diff
@@ -418,7 +437,15 @@ export function SearchResultGroupLayout<FilterOption>(
   }
 
   if (!resultItems?.length) {
-    return <>{noResultsComponent}</>;
+    if (noResultsComponent !== undefined) {
+      return <>{noResultsComponent}</>;
```

Comment:
If `noResultsComponent` is just a single element then is a fragment required here?
https://github.com/search?type=code&q=noResultsComponent++language%3ATSX+repo%3Abackstage%2Fbackstage

Reply:
@chalkedgoose Yes, because the return type has to be `JSX.Element` and not `ReactNode`

Reply:
Ah ok I don't want to drag a dead horse but I was under the impression those types were interchangeable in this instance but it looks like you can use primitives for noResultsComponent like strings and numbers.

Reply:
To make a long point short ReactNode may be the more appropriate option to return anyways and it also 
includes React Element as a subset 
https://github.com/facebook/react/blob/9b4f847d93aa302c953543ae0631023c06408ad3/packages/shared/ReactTypes.js#L11

Reply:
@chalkedgoose While that may pass for the current version of Backstage, React 18 requires that distinction to be made. If upgraded, TypeScript will complain about the return type unless things are narrowed and handled as in the above code. @schultzp2020 can verify this as we were both looking at the issue

Reply:
Good point do you mind sharing the link to these changes if you have free time 
i'd like to read more on it so I can keep up to date with React 18 changes.

Reply:
You're welcome to play around with my branch in the [React 18 upgrade PR](https://github.com/backstage/backstage/pull/18243) to see the issue for yourself.
I don't know where the actual definition changes are documented, so I wouldn't be able to tell you.

Reply:
@chalkedgoose Yeah, it's very weird behavior. I tried battling TS to return the correct type so we don't need an unnecessary fragment, but to no avail it did not work.